datacube/drivers/postgis/_api.py

```diff
@@ -103,49 +103,49 @@ def get_native_fields() -> dict[str, NativeField]:
         'id': NativeField(
             'id',
             'Dataset UUID',
-            Dataset.id
+            Dataset.__table__.c.id
         ),
         'indexed_time': NativeField(
             'indexed_time',
             'When dataset was indexed',
-            Dataset.added
+            Dataset.__table__.c.added
         ),
         'indexed_by': NativeField(
             'indexed_by',
             'User who indexed the dataset',
-            Dataset.added_by
+            Dataset.__table__.c.added_by
         ),
         'product': NativeField(
             'product',
             'Product name',
-            Product.name,
+            Product.__table__.c.name,
             join_clause=(Product.id == Dataset.product_ref)
         ),
         'product_id': NativeField(
             'product_id',
             'ID of a dataset type',
-            Dataset.product_ref
+            Dataset.__table__.c.product_ref
         ),
         'metadata_type': NativeField(
             'metadata_type',
             'Metadata type name of dataset',
-            MetadataType.name,
+            MetadataType.__table__.c.name,
             join_clause=(MetadataType.id == Dataset.metadata_type_ref),
         ),
         'metadata_type_id': NativeField(
             'metadata_type_id',
             'ID of a metadata type',
-            Dataset.metadata_type_ref
+            Dataset.__table__.c.metadata_type_ref
         ),
         'metadata_doc': NativeField(
             'metadata_doc',
             'Full metadata document',
-            Dataset.metadata_doc
+            Dataset.metadata_doc  # type: ignore[arg-type]
```

Comment:
I'm not sure what is going on with this, but changing it to the column like the other entries makes a lot of tests fail. Since there's apparently quite a lot of (at least implicit) test coverage for this, I  added the type ignore for now.